rust/parquet/src/schema/types.rs

```diff
@@ -103,6 +103,24 @@ impl Type {
         }
     }
 
+    /// Gets precision of this primitive type.
+    /// Note that this will panic if called on a non-primitive type.
+    pub fn get_precision(&self) -> i32 {
+        match *self {
+            Type::PrimitiveType { precision, ..} => precision,
+            _ => panic!("Cannot call get_precision() on non-primitive type")
+        }
+    }
+
+    /// Gets scale of this primitive type.
+    /// Note that this will panic if called on a non-primitive type.
+    pub fn get_scale(&self) -> i32 {
+        match *self {
+            Type::PrimitiveType { scale, ..} => scale,
+            _ => panic!("Cannot call get_scale() on non-primitive type")
```

Comment:
this is much nicer (and it is now clearer what is expected) 👍

Reply:
I added one more commit, using the new getters in `array_reader` to simplify the decimal converter construction.